src/Components/Server/src/BlazorPack/BlazorPackHubProtocolWorker.cs

```diff
@@ -35,6 +35,13 @@ protected override object DeserializeObject(ref MessagePackReader reader, Type t
                 {
                     return reader.ReadSingle();
                 }
+                else if (type == typeof(ReadOnlySequence<byte>))
+                {
+                    // This is how I think it should work, but sometimes the memory seems to get corrupted.
+                    // The "await foreach (var chunk in subject)" code in RemoteJSDataStream sometimes gets
+                    // chunks with negative lengths. So, this is not actually used in this proof-of-concept
+                    return reader.ReadBytes() ?? null;
```

Comment:
I haven't been able to repro negative chunk length, @SteveSandersonMS was there a particular scenario in which this popped up?

Reply:
We should for the time being use byte arrays (as we know that works) and look at optimizing for ReadOnlySequence<byte> on all locations in the future.

We can avoid the byte array/string allocation across the board on the circuit host if we handle it here, and it makes sense for us to tackle it everywhere and separately.

Reply:
Limiting it to `byte[]` sounds fine to me unless @TanayParikh thinks that the behavior of `ReadOnlySequence<byte>` is going to be well understood and we can already rely on that.